src/sql/parts/dashboard/tabs/dashboardWidgetTab.component.ts

```diff
@@ -5,123 +5,55 @@
 
 import 'vs/css!./dashboardWidgetTab';
 
-import { Component, Inject, Input, forwardRef, ViewChild, ElementRef, ViewChildren, QueryList, OnDestroy, ChangeDetectorRef, EventEmitter, OnChanges } from '@angular/core';
+import { Component, Inject, Input, forwardRef, ViewChild, ElementRef, ViewChildren, QueryList, OnDestroy, ChangeDetectorRef, EventEmitter, OnChanges, AfterContentInit } from '@angular/core';
 import { NgGridConfig, NgGrid, NgGridItem } from 'angular2-grid';
 
 import { DashboardServiceInterface } from 'sql/parts/dashboard/services/dashboardServiceInterface.service';
 import { TabConfig, WidgetConfig } from 'sql/parts/dashboard/common/dashboardWidget';
 import { DashboardWidgetWrapper } from 'sql/parts/dashboard/common/dashboardWidgetWrapper.component';
 import { subscriptionToDisposable } from 'sql/base/common/lifecycle';
 import { DashboardTab } from 'sql/parts/dashboard/common/interfaces';
+import { WidgetTabContent } from 'sql/parts/dashboard/tabContents/widgetTabContent.component';
 
 import { Disposable, IDisposable } from 'vs/base/common/lifecycle';
 import { ConfigurationTarget } from 'vs/platform/configuration/common/configuration';
 import * as objects from 'vs/base/common/objects';
 import Event, { Emitter } from 'vs/base/common/event';
 
-/**
- * Sorting function for dashboard widgets
- * In order of priority;
- * If neither have defined grid positions, they are equivalent
- * If a has a defined grid position and b does not; a should come first
- * If both have defined grid positions and have the same row; the one with the smaller col position should come first
- * If both have defined grid positions but different rows (it doesn't really matter in this case) the lowers row should come first
- */
-function configSorter(a, b): number {
-	if ((!a.gridItemConfig || !a.gridItemConfig.col)
-		&& (!b.gridItemConfig || !b.gridItemConfig.col)) {
-		return 0;
-	} else if (!a.gridItemConfig || !a.gridItemConfig.col) {
-		return 1;
-	} else if (!b.gridItemConfig || !b.gridItemConfig.col) {
-		return -1;
-	} else if (a.gridItemConfig.row === b.gridItemConfig.row) {
-		if (a.gridItemConfig.col < b.gridItemConfig.col) {
-			return -1;
-		}
-
-		if (a.gridItemConfig.col === b.gridItemConfig.col) {
-			return 0;
-		}
-
-		if (a.gridItemConfig.col > b.gridItemConfig.col) {
-			return 1;
-		}
-	} else {
-		if (a.gridItemConfig.row < b.gridItemConfig.row) {
-			return -1;
-		}
-
-		if (a.gridItemConfig.row === b.gridItemConfig.row) {
-			return 0;
-		}
-
-		if (a.gridItemConfig.row > b.gridItemConfig.row) {
-			return 1;
-		}
-	}
-
-	return void 0; // this should never be reached
-}
-
 @Component({
 	selector: 'dashboard-widget-tab',
-	templateUrl: decodeURI(require.toUrl('sql/parts/dashboard/tabs/dashboardWidgetTab.component.html')),
-	providers: [{ provide: DashboardTab, useExisting: forwardRef(() => DashboardWidgetTab) }]
```

Comment:
Also dashboardWidgetWrapper.component.html. do we still need that file?

Reply:
yes we do need this file

---
In reply to: [168604494](https://github.com/Microsoft/sqlopsstudio/pull/716#discussion_r168604494) [](ancestors = 168604494)